System/Random.hs

```diff
@@ -222,11 +244,46 @@ class Monad m => MonadRandom g m where
   uniformWord32 = uniformWord32R maxBound
   uniformWord64 :: g -> m Word64
   uniformWord64 = uniformWord64R maxBound
+  uniformBytes :: Int -> g -> m ByteArray
+  default uniformBytes :: PrimMonad m => Int -> g -> m ByteArray
+  uniformBytes = uniformByteArrayPrim
+  {-# INLINE uniformBytes #-}
+
+
+-- | This function will efficiently generate a sequence of random bytes in a platform
+-- independent manner. Memory allocated will be pinned, so it is safe to use for FFI
+-- calls.
+uniformByteArrayPrim :: (MonadRandom g m, PrimMonad m) => Int -> g -> m ByteArray
+uniformByteArrayPrim n0 gen = do
+  let n = max 0 n0
+      (n64, nrem64) = n `quotRem` 8
+  ma <- newPinnedByteArray n
+  let go i ptr
+        | i < n64 = do
+          w64 <- uniformWord64 gen
+          unsafeIOToPrim $ runF word64LE w64 ptr
+          go (i + 1) (ptr `plusPtr` 8)
+        | otherwise = return ptr
+  ptr <- go 0 (mutableByteArrayContentsCompat ma)
+  when (nrem64 > 0) $ do
+    w64 <- uniformWord64 gen
+    -- in order not to mess up the byte order we write generated Word64 into a temporary
+    -- pointer and then copy only the missing bytes over to the array
```

Comment:
A previous iteration of this code used `uniformWord8` to generate those remaining bytes. Did you replace the `uniformWord8` based generation for the remaining bytes by `uniformWord64` based generation because of byte order?

If that is the case, I think this comment could be even more explicit: "... we write generated Word64 _instead of generating Word8 one by one_ ..." or something like that.

Reply:
It was not only byte order. Generating 1xWord64 and 8x Word8 concatenated together will always give you different result, because in order generate 8x Word8 we need to generate 8x Word64 first (at least in case of splitmix)

I'll add a more descriptive comment to the function